node/pkg/watchers/sui/watcher.go

```diff
@@ -503,25 +420,141 @@ func (e *Watcher) Run(ctx context.Context) error {
 
 	select {
 	case <-ctx.Done():
-		_ = ws.Close(websocket.StatusNormalClosure, "")
 		return ctx.Err()
 	case err := <-errC:
-		_ = ws.Close(websocket.StatusInternalError, err.Error())
 		return err
 	}
 }
 
-// fixSuiWsURL ensures the websocket scheme is properly specified
-func (e *Watcher) fixSuiWsURL(logger *zap.Logger) error {
-	u, _ := url.Parse(e.suiWS)
+func (w *Watcher) getEvents() (SuiEventResponseData, error) {
+	// Only get events newer than the last processed height
+	var retVal SuiEventResponseData
+	var nextCursor struct {
+		TxDigest string
+		EventSeq string
+	}
+	firstTime := true
+	done := false
+	for !done {
+		var reader io.Reader
+		if firstTime {
+			reader = strings.NewReader(
+				fmt.Sprintf(`{"jsonrpc":"2.0", "id": 1, "method": "suix_queryEvents", "params": [{ "MoveEventType": "%s" }, null, %d, %t]}`,
```

Comment:
Since this string doesn't change, could you create it at start up and store it in w, so you don't have to do it every time?

Reply:
Changed